src/main/java/com/powsybl/openloadflow/ac/PhaseControlOuterLoop.java

```diff
@@ -34,6 +38,47 @@ public String getType() {
         return "Phase control";
     }
 
+    @Override
+    public void initialize(LfNetwork network) {
+        List<LfBranch> controllerBranches = new ArrayList<>(1);
+        List<LfBranch> disabledBranches = new ArrayList<>(1);
+        for (LfBranch branch : network.getBranches()) {
+            if (branch.isPhaseController() && branch.isPhaseControlEnabled()) {
+                controllerBranches.add(branch);
+            }
+            if (branch.isDisabled()) {
+                disabledBranches.add(branch);
+            }
+        }
+        if (!controllerBranches.isEmpty()) {
+            for (LfBranch controllerBranch : controllerBranches) {
+                var phaseControl = controllerBranch.getDiscretePhaseControl().orElseThrow();
+                var controlledBranch = phaseControl.getControlled();
+                var connectivity = network.getConnectivity();
+
+                // apply contingency (in case we are inside a security analysis)
+                for (LfBranch disabledBranch : disabledBranches) {
+                    connectivity.cut(disabledBranch.getBus1(), disabledBranch.getBus2());
+                }
+                int smallComponentsCountBeforePhaseShifterLoss = connectivity.getSmallComponents().size();
+
+                // then the phase shifter controlled branch
+                if (!disabledBranches.contains(controlledBranch)) {
```

Comment:
I think it was not the correct fix, as when a phase shifter is disable it is already deactivated (see ac equation system), so we can just just skip the connectivity check.